File: algorithms/lab3/bubble_sort/bubble_sort.py
from random import randint


def bubble_sort(data):
    for i in range(len(data)):
        for j in range(len(data) - 1, i, -1):
            if data[j] < data[j - 1]:
                data[j], data[j - 1] = data[j - 1], data[j]

    return data


if __name__ == "__main__":
    data = [randint(0, 100) for i in range(10)]
    print('Before:', data)
    print('After:', bubble_sort(data))

File: algorithms/lab3/bubble_sort/bubble_sort.js
function bubble_sort(data){
    let sorted = false;
    while (!sorted) {
        sorted = true;
        data.forEach(function (element, index, array){
            if (element > array[index+1]) {
                array[index] = array[index + 1];
                array[index + 1] = element;
                sorted = false;
      }
    });
  }
  return data;
}

const data = [7, 4, 3, 8, 2, 1, 5, 9];
console.log('Before:', data);
console.log('After:', bubble_sort(data));
